File: src/main/java/no/fintlabs/provider/config/KafkaConfig.java
package no.fintlabs.provider.config;

import lombok.Getter;
import lombok.extern.slf4j.Slf4j;
import org.apache.commons.lang3.RandomStringUtils;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.env.Environment;

@Getter
@Slf4j
@Configuration
public class KafkaConfig {

    public static final int SUFFIX_LENGTH = 8;
    private final String groupIdSuffix;

    public KafkaConfig(Environment environment) {
        groupIdSuffix = RandomStringUtils.random(SUFFIX_LENGTH, true, true).toLowerCase();

        String groupId = environment.getProperty("spring.kafka.consumer.group-id");
        log.info("Group-id: %s-%s".formatted(groupId, groupIdSuffix));
    }

}

File: src/main/java/no/fintlabs/provider/config/KafkaConfig.kt
package no.fintlabs.provider.config

import org.apache.commons.lang3.RandomStringUtils
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration

@Configuration
open class KafkaConfig(
    @Value($$"${spring.kafka.consumer.group-id:}") groupId: String
) {

    private val logger = LoggerFactory.getLogger(javaClass)
    val groupIdSuffix: String = createGroupIdSuffix(groupId)

    private fun createGroupIdSuffix(groupId: String): String {
        val suffix = RandomStringUtils.secure().nextAlphanumeric(SUFFIX_LENGTH).lowercase()
        logger.info("Group-id: $groupId-$suffix")
        return suffix
    }

    companion object {
        const val SUFFIX_LENGTH: Int = 8
    }
}
